types/node/index.d.ts

```diff
@@ -1852,9 +1852,22 @@ declare module "url" {
         hash?: string;
         path?: string;
     }
+    
+    export interface FormatOptions {
+        protocol?: string;
+        slashes?: boolean;
+        auth?: string;
+        host?: string;
+        hostname?: string;
+        port?: string | number;
+        pathname?: string;
+        search?: string;
+        query?: { [key: string]: any; };
+        hash?: string;
+    }
 
     export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
-    export function format(url: Url): string;
+    export function format(urlObject: FormatOptions): string;
```

Comment:
In NodeJS 7 there is an additional variant to parse WHATWG URL objects. see https://nodejs.org/dist/latest-v7.x/docs/api/url.html#url_url_format_url_options
I think this should be also added here for completeness.